akka-sample-kafka-to-sharding/processor/src/main/scala/sample/sharding/kafka/UserEvents.scala

```diff
@@ -0,0 +1,77 @@
+package sample.sharding.kafka
+
+import akka.Done
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.external.ExternalShardAllocationStrategy
+import akka.cluster.sharding.typed.ClusterShardingSettings
+import akka.cluster.sharding.typed.Murmur2NoEnvelopeMessageExtractor
+import akka.cluster.sharding.typed.scaladsl.ClusterSharding
+import akka.cluster.sharding.typed.scaladsl.Entity
+import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+
+object UserEvents {
+
+  val TypeKey: EntityTypeKey[UserEvents.Message] =
+    EntityTypeKey[UserEvents.Message]("user-processing")
+
+  sealed trait Message {
+    def userId: String
+  }
+  sealed trait UserEvent extends Message with CborSerializable
+  case class UserAction(userId: String, description: String, replyTo: ActorRef[Done]) extends UserEvent
+  case class UserPurchase(userId: String, product: String, quantity: Long, priceInPence: Long, replyTo: ActorRef[Done])
+      extends UserEvent
+
+  sealed trait UserQuery extends Message
+  case class GetRunningTotal(userId: String, replyTo: ActorRef[RunningTotal]) extends UserQuery
+
+  case class RunningTotal(totalPurchases: Long, amountSpent: Long)
+
+  def apply(userId: String): Behavior[Message] = running(RunningTotal(0, 0))
```

Comment:
What is the purpose of the `userId` parameter?

Reply:
I'd just copied that from the docs for typed cluster sharding, we don't need the entity id here as it is in the message so removing